wagtail/contrib/search_promotions/views.py

```diff
@@ -2,85 +2,104 @@
+from wagtail.admin.views.generic import IndexView as WagtailIndexView
 from wagtail.contrib.search_promotions import forms, models
 from wagtail.contrib.search_promotions.models import Query
 from wagtail.log_actions import log
 from wagtail.search.utils import normalise_query_string
 
+decorators = [vary_on_headers, any_permission_required]
 
-@any_permission_required(
-    "wagtailsearchpromotions.add_searchpromotion",
-    "wagtailsearchpromotions.change_searchpromotion",
-    "wagtailsearchpromotions.delete_searchpromotion",
-)
-@vary_on_headers("X-Requested-With")
-def index(request):
-    # Ordering
-    valid_ordering = ["query_string", "-query_string", "views", "-views"]
-    ordering = valid_ordering[0]
-
-    if "ordering" in request.GET and request.GET["ordering"] in valid_ordering:
-        ordering = request.GET["ordering"]
-
-    # Query
-    queries = Query.objects.filter(editors_picks__isnull=False).distinct()
-
-    if "views" in ordering:
-        queries = queries.annotate(views=functions.Coalesce(Sum("daily_hits__hits"), 0))
 
-    queries = queries.order_by(ordering)
-
-    # Search
+@method_decorator(
+    any_permission_required(
+        "wagtailsearchpromotions.add_searchpromotion",
+        "wagtailsearchpromotions.change_searchpromotion",
+        "wagtailsearchpromotions.delete_searchpromotion",
+    ),
+    name="dispatch",
+)
+@method_decorator(vary_on_headers("X-Requested-With"), name="dispatch")
+class IndexView(WagtailIndexView):
+    template_name = "wagtailsearchpromotions/index.html"
+    result_template_name = "wagtailsearchpromotions/results.html"
+    paginate_by = 20
     is_searching = False
-    query_string = request.GET.get("q", "")
-
-    if query_string:
-        queries = queries.filter(query_string__icontains=query_string)
-        is_searching = True
-
-    # Paginate
-    paginator = Paginator(queries, per_page=20)
-    try:
-        queries = paginator.page(request.GET.get("p", 1))
-    except InvalidPage:
-        raise Http404
-
-    if request.headers.get("x-requested-with") == "XMLHttpRequest":
-        return TemplateResponse(
-            request,
-            "wagtailsearchpromotions/results.html",
-            {
-                "is_searching": is_searching,
-                "ordering": ordering,
-                "queries": queries,
-                "query_string": query_string,
-            },
-        )
-    else:
-        return TemplateResponse(
-            request,
-            "wagtailsearchpromotions/index.html",
-            {
-                "is_searching": is_searching,
-                "ordering": ordering,
-                "queries": queries,
-                "query_string": query_string,
-                "search_form": SearchForm(
-                    data={"q": query_string} if query_string else None,
-                    placeholder=_("Search promoted results"),
-                ),
-            },
-        )
+    ordering = "query_string"
```

Comment:
```suggestion
    ordering = "query_string"
    model = Query
```

We can use the `model` attribute to make a lot of the other code easier to work with as the CBVs will build out most queries for us.